site/app/controllers/forum/ForumController.php

```diff
@@ -786,6 +786,12 @@ public function showThreads($thread_id = null, $option = 'tree'){
         if(!empty($posts)){
             $thread_id = $posts[0]["thread_id"];
         }
+        foreach($posts as &$post) {
+            $count = 1;
+            while($count > 0) {
+                $post['content'] = preg_replace('/(?:!\[(.*?)\]\((.*?)\))/', '$2', $post['content'], -1, $count);
+            }
+        }
```

Comment:
Use the little thought of do-while here for a bit more readability in the sequence of what you're doing.
```
do {
    $post['content'] = preg_replace('/(?:!\[(.*?)\]\((.*?)\))/', '$2', $post['content'], -1, $count);
} while ($count > 0);
```

Reply:
fixed